File: rust/657.robot-return-to-origin.rs
/*
 * @lc app=leetcode id=657 lang=rust
 *
 * [657] Robot Return to Origin
 */

// @lc code=start
impl Solution {
    pub fn judge_circle(moves: String) -> bool {
        let mut position = (0, 0);
        for d in moves.chars() {
            match d {
                'U' => position.0 -= 1,
                'R' => position.1 += 1,
                'D' => position.0 += 1,
                _ => position.1 -= 1,
            }
        }
        position == (0, 0)
    }
}
// @lc code=end

File: javascript/657.robot-return-to-origin.js
/*
 * @lc app=leetcode id=657 lang=javascript
 *
 * [657] Robot Return to Origin
 */

// @lc code=start
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function (moves) {
  let x = 0,
    y = 0;
  for (const d of moves) {
    if (d === "U") y--;
    else if (d === "R") x++;
    else if (d === "D") y++;
    else x--;
  }
  return x === 0 && y === 0;
};
// @lc code=end
